### PR TITLE
#2094. Remove TODO, update assertions in Language/Mixin tests

### PR DESCRIPTION
**Pull request by sgrekhov**
Current PR
- removed TODO
- updated assertions

Upcoming changes:
- rename and reorder files according to the current spec structure
- add missing test

### CODE REVIEW

File: Language/Mixins/Mixin_Application/abstract_t01.dart [hunk 1]
```diff
@@ -2,25 +2,24 @@
...
-/// the scope of L, bound to the class defined by the mixin application M.
-/// The name of the class is also set to C. Iff the class is prefixed by the
-/// built-in identifier abstract, the class being defined is an abstract class.
-/// @description Checks that class C is not abstract (can be instantiated by
-/// new) if mixin application is not abstract and there are no abstract
-/// identifier
+/// @assertion Let D be a mixin application class declaration of the form
+/// abstract? class N = S with M1, ..., Mn implements I1, ..., Ik;
+/// ...
+  /// The effect of D in library L is to introduce the name N into the scope of
```
Comment (chloestefantsova) [inline Language/Mixins/Mixin_Application/abstract_t01.dart:8]: Should that extra space before the comment be removed?

File: Language/Mixins/Mixin_Application/interfaces_t01.dart [hunk 2]
```diff
@@ -2,26 +2,25 @@
...
-/// @assertion If the mixin application declares support for interfaces, the
-/// resulting class implements those interfaces.
-/// @description Checks that if the mixin application declares support for
-/// interfaces and the resulting class implements those interfaces then no
-/// static warning occurs
+/// @assertion Let D be a mixin application class declaration of the form
+/// abstract? class N = S with M1, ..., Mn implements I1, ..., Ik;
+/// ...
+// The effect of D in library L is to introduce the name N into the scope of L,
+// bound to the class defined by the clause S with M1, ..., Mn with name N,
```
Comment (chloestefantsova) [inline Language/Mixins/Mixin_Application/interfaces_t01.dart:9]: Should these two lines be a doc-comment starting with `///` as the comments above and below it?

File: Language/Mixins/Mixin_Application/interfaces_t02.dart [hunk 3]
```diff
@@ -2,19 +2,17 @@
...
-/// resulting class implements those interfaces.
-/// @description Checks that if the mixin application declares support for
-/// interfaces and the resulting class implements those interfaces then no
-/// static warning occurs. Interface num get g => 0; declared in class S of the
-/// S with M
+/// @assertion Let D be a mixin application class declaration of the form
+/// abstract? class N = S with M1, ..., Mn implements I1, ..., Ik;
+/// ...
+// The effect of D in library L is to introduce the name N into the scope of L,
+// bound to the class defined by the clause S with M1, ..., Mn with name N,
```
Comment (chloestefantsova) [inline Language/Mixins/Mixin_Application/interfaces_t02.dart:9]: Ditto.

File: Language/Mixins/Mixin_Application/interfaces_t03.dart [hunk 4]
```diff
@@ -2,26 +2,26 @@
...
-/// @assertion If the mixin application declares support for interfaces, the
-/// resulting class implements those interfaces.
-/// @description Checks that if the mixin application declares support for
-/// interfaces and the resulting class implements those interfaces then no
-/// static warning occurs. Interface implementation declared in mixin
+/// @assertion Let D be a mixin application class declaration of the form
+/// abstract? class N = S with M1, ..., Mn implements I1, ..., Ik;
+/// ...
+// The effect of D in library L is to introduce the name N into the scope of L,
+// bound to the class defined by the clause S with M1, ..., Mn with name N,
```
Comment (chloestefantsova) [inline Language/Mixins/Mixin_Application/interfaces_t03.dart:9]: Ditto.

File: Language/Mixins/Mixin_Application/interfaces_t04.dart [hunk 5]
```diff
@@ -2,19 +2,18 @@
...
-/// resulting class implements those interfaces.
-/// @description Checks that if the mixin application declares support for
-/// interfaces and the resulting class implements those interfaces then no
-/// static warning occurs. Interface implementation declared partially in a
-/// mixin, partially in a superclass
+/// @assertion Let D be a mixin application class declaration of the form
+/// abstract? class N = S with M1, ..., Mn implements I1, ..., Ik;
+/// ...
+// The effect of D in library L is to introduce the name N into the scope of L,
+// bound to the class defined by the clause S with M1, ..., Mn with name N,
```
Comment (chloestefantsova) [inline Language/Mixins/Mixin_Application/interfaces_t04.dart:9]: Ditto.

File: Language/Mixins/Mixin_Application/interfaces_t05.dart [hunk 6]
```diff
@@ -2,22 +2,26 @@
...
-/// @assertion If the mixin application declares support for interfaces, the
-/// resulting class implements those interfaces.
-/// @description Checks that if the mixin application declares support for
-/// interfaces and the resulting class does not implement those interfaces then
-/// compile error occurs
+/// @assertion Let D be a mixin application class declaration of the form
+/// abstract? class N = S with M1, ..., Mn implements I1, ..., Ik;
+/// ...
+// The effect of D in library L is to introduce the name N into the scope of L,
+// bound to the class defined by the clause S with M1, ..., Mn with name N,
```
Comment (chloestefantsova) [inline Language/Mixins/Mixin_Application/interfaces_t05.dart:9]: Ditto.

File: Language/Mixins/Mixin_Application/interfaces_t06.dart [hunk 7]
```diff
@@ -2,26 +2,25 @@
...
-/// @assertion If the mixin application declares support for interfaces, the
-/// resulting class implements those interfaces.
-/// @description Checks that if the mixin application declares support for
-/// interfaces and the resulting class implements those interfaces then no
-/// static warning occurs. Test type aliases
+/// @assertion Let D be a mixin application class declaration of the form
+/// abstract? class N = S with M1, ..., Mn implements I1, ..., Ik;
+/// ...
+// The effect of D in library L is to introduce the name N into the scope of L,
+// bound to the class defined by the clause S with M1, ..., Mn with name N,
```
Comment (chloestefantsova) [inline Language/Mixins/Mixin_Application/interfaces_t06.dart:9]: Ditto.

File: Language/Mixins/Mixin_Application/interfaces_t07.dart [hunk 8]
```diff
@@ -2,19 +2,18 @@
...
-/// resulting class implements those interfaces.
-/// @description Checks that if the mixin application declares support for
-/// interfaces and the resulting class implements those interfaces then no
-/// static warning occurs. Interface implementation declared partially in a
-/// mixin, partially in a superclass. Test type alias
+/// @assertion Let D be a mixin application class declaration of the form
+/// abstract? class N = S with M1, ..., Mn implements I1, ..., Ik;
+/// ...
+// The effect of D in library L is to introduce the name N into the scope of L,
+// bound to the class defined by the clause S with M1, ..., Mn with name N,
```
Comment (chloestefantsova) [inline Language/Mixins/Mixin_Application/interfaces_t07.dart:9]: Ditto.

File: Language/Mixins/declaring_constructor_t11.dart [hunk 9]
```diff
@@ -2,12 +2,16 @@
...
 
-/// @assertion It is a compile-time error if a declared or derived mixin
-/// explicitly declares a constructor which is not a factory constructor.
+/// @assertion It is a compile-time error to derive a mixin from a class that
+/// declares a generative constructor, or from a class that has a superclass
+/// other than Object.
+///
 /// @description Checks that it is a compile-time error if a derived mixin
-/// explicitly declares a constructor.
+/// explicitly declares a constructor. Test type alias
```
Comment (chloestefantsova) [inline Language/Mixins/declaring_constructor_t11.dart:10]: This looks like an incomplete sentence.

**Review by chloestefantsova — CHANGES_REQUESTED** [review]

**Comment by sgrekhov** [comment]
Updated. Please take another look

**Review by chloestefantsova — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**